src/commands/mount.rs

```diff
@@ -0,0 +1,101 @@
+//! `mount` subcommand
+mod fusefs;
+
+use std::{ffi::OsStr, path::PathBuf};
+
+use crate::{repository::CliIndexedRepo, status_err, Application, RUSTIC_APP};
+
+use abscissa_core::{Command, Runnable, Shutdown};
+use anyhow::Result;
+use fuse_mt::{mount, FuseMT};
+use rustic_core::vfs::{IdenticalSnapshot, Latest, Vfs};
+
+use fusefs::FuseFS;
+
+#[derive(clap::Parser, Command, Debug)]
+pub(crate) struct MountCmd {
+    /// The path template to use for snapshots. {id}, {id_long}, {time}, {username}, {hostname}, {label}, {tags}, {backup_start}, {backup_end} are replaced. [default: "[{hostname}]/[{label}]/{time}"]
+    #[clap(long)]
+    path_template: Option<String>,
+
+    /// The time template to use to display times in the path template. See https://docs.rs/chrono/latest/chrono/format/strftime/index.html for format options. [default: "%Y-%m-%d_%H-%M-%S"]
+    #[clap(long)]
+    time_template: Option<String>,
+
+    #[clap(value_name = "PATH")]
+    /// The mount point to use
+    mountpoint: PathBuf,
+
+    /// Specify directly which path to mount
+    #[clap(value_name = "SNAPSHOT[:PATH]")]
+    snap: Option<String>,
+
+    /// Don't allow other users to access the mount point
+    #[clap(long)]
+    no_allow_other: bool,
+}
+
+impl Runnable for MountCmd {
+    fn run(&self) {
+        if let Err(err) = RUSTIC_APP
+            .config()
+            .repository
+            .run_indexed(|repo| self.inner_run(repo))
+        {
+            status_err!("{}", err);
+            RUSTIC_APP.shutdown(Shutdown::Crash);
+        };
+    }
+}
+
+impl MountCmd {
+    fn inner_run(&self, repo: CliIndexedRepo) -> Result<()> {
+        let config = RUSTIC_APP.config();
+
+        let path_template = self
+            .path_template
+            .clone()
+            .unwrap_or_else(|| "[{hostname}]/[{label}]/{time}".to_string());
+        let time_template = self
+            .time_template
+            .clone()
+            .unwrap_or_else(|| "%Y-%m-%d_%H-%M-%S".to_string());
```

Comment:
Is that something we want to be able to override from config?

Reply:
Is added. I just wonder how we want to use this in our config tests.. Should we add this to `full.toml` and the `check-config` fixture?

BTW: We have the same topic with the "webdav" feature, but this we include in all our tests. Here, we have a feature which is definitively not included in some targets, like windows...

Or maybe it is a better idea to compile the config structure unconditionally and just have the command conditionally compiled?

Reply:
Hmm, the question is what happens if a config value is given on a system that doesn't support this, e.g. webdav or mount. Is there an error like 'The webdav feature is not supported on your system, it will be ignored in the config.'or something? Because that would be useful I guess and good UX.

Reply:
You get an error when parsing the config, e.g. when compiled without the `mount` feature:
```
error: rustic-rs fatal error: parse error: unknown field `mount`, expected one of `global`, `repository`, `snapshot-filter`, `backup`, `copy`, `forget`, `webdav` at line 32 column 1
```

I actually think we should exactly use this behavior. If there is no `mount` it is simply not wanted in the config. 

About tests I think we should add this to the `full.toml` and `show-config` tests - as soon as we add the `mount` option to the tests... (maybe this requires to use special test fixtures for windows...)

Reply:
Do we have control over the error message? Because that one is pretty bad and doesn't say anything what is happening and why this is an unknown field in this case. We need to document this error somewhere, so we don't possibly get a lot of issues about this.

Also, I'm not sure about the behaviour itself. Someone could migrate their config to another OS. E.g. from Linux to another Mac or Windows system, where they wouldn't be currently able to use mount. And then suddenly they get this error. Is there a way to check if the error is regarding `mount` or `webdav` and if so, spit out a specialized message, in regard to the missing feature on that platform or so?

Reply:
Mhh, ok, if we want to support copying tomls over, I agree that shouldn't be an error...
But then I would not even print a warning - if we use another command when having compiled with the `mount` option, that part of the config is also not used and we don't warn about that..

Reply:
I think we need to decide if the profile is OS-dependent or not. I would root for `not`, because it makes it easier to test as well as writing examples for the whole user base.

Instead of `deny_unknown_fields` I would propose giving feedback to the user, that certain keys are in the profile and not being used for whatever reason. e.g. `mount` and `webdav` if support is not compiled in, or if keys are used, that are not valid in a profile context.

I think that is pretty standard procedure to warn about content in a profile that has no effect, to not leave the user in the dark about it. As they may think, that something is set, and derive the current behaviour from that, although it's not even being considered. Rclone options come to mind and others. So I think it's valuable to warn about that.

Maybe something like: https://crates.io/crates/serde_path_to_error can be helpful here.

Reply:
I agree we should just keep the feature flag.

About the warning: In principle I agree about general options, but here we have options specific for a command which cannot even be called when not compiled with the feature...
I wonder how we should be able to print a warning. Imagine a user running `rustic backup` on a version without the `mount` feature. Should we warn "you have [mount] in your profile which is not used"? I mean when users call `rustic mount` with that build, they will already get an error and `[mount]` is only ever used in the `mount` command...
And users using a version with the `mount` feature would call `rustic backup` without getting a warning about `[mount]` in the profile - even though their called command does not use any of the options set there...?!?

I have now added a `Value` field in the case without the feature - that will suppress the error but also don't print any warning...

Reply:
I'm thinking more of a `println` than a warning I think, as an `INFO`. But independent of the logging. So users become aware of that and can remove it manually from the profile if they want.

Reply:
That being said, this is just the way I think it makes sense to do it, for informing the user, about values in the config, that have no effect. We don't need to do it that way. Also, we may need to refactor some things, regarding the config parsing step for this, to be able to spit out better errors/warnings? So probably nothing for this PR here.